System/Console/ANSI/Types.hs

```diff
@@ -26,11 +26,29 @@ data Color = Black
            | White
            deriving (Eq, Ord, Bounded, Enum, Show, Read, Ix)
 
--- | ANSI colors come in two intensities
+-- | ANSI's standard colors come in two intensities.
 data ColorIntensity = Dull
                     | Vivid
                     deriving (Eq, Ord, Bounded, Enum, Show, Read, Ix)
 
+-- | Represents xterm's protocol for a 256-color palette.
+--
+-- The protocol has three parts:
+--
+-- (1) the eight standard, or 'system', colors in dull and vivid intensities;
+--
+-- (2) colors in RGB space, with the red, green and blue color components being
+-- specified by a integer from 0 to 5; and
+--
+-- (3) a spectrum of 24 grays, from dark gray to near white (black and white are
+-- themselves excluded).
+data Xterm256Color
+  = SystemColor !ColorIntensity !Color -- ^ The ANSI standard colors
+  | RGBColor !(Colour Float)           -- ^ Colors in RGB space
```

Comment:
Now that this type changed from a discrete 6x6x6 cube to a continuous 1x1x1 cube, I think it's important to state in the documentation that this continuous representation of a color, taking it components as a float between 0 and 1, will be rounded to the nearest xterm rgb color. So that people don't think that "just because the type allows for many colors, that they are all supported".

Before it was implicit from the type we used, but now it's more "vague"...

Reply:
My new revised approach involves a 'constructor' for a discrete 6x6x6 cube:
~~~~
xterm6LevelRGB :: Int -> Int -> Int -> Word8

[ SetPaletteColor Background $ xterm6LevelRGB 1 2 3 ]
~~~~